### PR TITLE
To provide some boundaries to enable easier testing of middleware/route configuration applicators

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: src/Dilex.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class Dilex extends Application implements RouteContainer, MiddlewareContainer
+{
+    /**
+     * @param string $pattern
+     * @param Endpoint $to
+     * @return \Silex\Controller
+     */
+    public function get( $pattern, $to = null )
+    {
+        return parent::get( $pattern, $to );
```
Comment (adrianclay) [inline src/Dilex.php:17]: Why do you override the methods?

**Comment by adrianclay** [comment]
Boundaries++

File: src/Dilex.php (hunk 1 above)
Comment (craigjbass) [inline src/Dilex.php:17]: So I can overide the typehint on the $to to ``Endpoint`

File: src/Dilex.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+namespace Clearbooks\Dilex;
+
+use Silex\Application;
+
+class Dilex extends Application implements RouteContainer, MiddlewareContainer
+{
+    /**
+     * @param string $pattern
+     * @param Endpoint $to
```
Comment (craigjbass) [inline src/Dilex.php:12]: These type hints are wrong, should be string. Or rather a ::class of type Endpoint

**Pull request merged**